libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1557,6 +1543,16 @@ void IRGeneratorForStatements::endVisit(FunctionCallOptions const& _options)
 void IRGeneratorForStatements::endVisit(MemberAccess const& _memberAccess)
 {
 	setLocation(_memberAccess);
+
+	FunctionDefinition const* addToDispatchFunction = nullptr;
+
+	// To make sure we don't forget any scenario we call this using a
+	// scopeguard.
+	ScopeGuard addToInternalDispatch{[&] {
```

Comment:
I don't really see the benefit of the scope guard - if we forget to assign to addToDispatchFunction the effect is the same. You might change this to a local lambda function that adds the check about `calledDirectly`.